Automattic/jetpack
# Refactor forms dashboard state to not rely on wp.data resolvers

## Proposed changes:

This patch refactors the state management for the forms dashboard. Since we don't want to cache the results anyway, it makes more sense to remove the resolvers and simply trigger the request every time the query changes rather than manually invalidating them every time, needlessly increasing complexity.

I also went ahead and fixed the state to work with the latest `totals` property on the response so paging works correctly again.

### Other information:


## Does this pull request change what data or activity we track or use?



No.

## Testing instructions:






- Go to `/wp-admin/admin.php?page=jetpack-forms`.
- Your feedback results should still load correctly.
- You should be able to switch between tabs (All, Spam, Trash) and the results should reload as expected.
- Search should still work as expected.
